core-api/api_container_service.proto

```diff
@@ -165,3 +168,16 @@ message ExecCommandResponse {
   int32 exit_code = 1;
   bytes log_output = 2;
 }
+
+// ==============================================================================================
+//                                     Check Availability
+// ==============================================================================================
+message CheckAvailabilityArgs {
+  string service_id = 1;
+  string port = 2;
+  string path = 3;
+  int32 initial_delay_seconds = 4;
```

Comment:
Also recommend using `uint` rathe rthan `int` for these variables, since it doesn't make sense for them to be negative

Reply:
Great suggest!